modules/pel/peltool/peltool.py

```diff
@@ -188,38 +256,65 @@ def main():
     parser.add_argument('-n', '--non-serviceable',
                         help='Only parse non-serviceable (info/recovered) PELs',
                         action='store_true')
+    parser.add_argument('-p', '--no-plugins', dest='skip_plugins',
+                        action='store_true', help='Skip loading plugins')
+    parser.add_argument('-d', '--directory', dest='directory',
+                        help='Process all files in a directory and save as <filename>.json.'
+                        ' Use -o to specify output directory')
+    parser.add_argument('-o', '--output-dir', dest='output_dir',
+                        help='Directory to write output files when processing a directory')
+    parser.add_argument('-e', '--extension', dest='extension',
+                        help='Used with -d, only look for files with this extension (e.g. ".pel")')
+    parser.add_argument('-x', '--delete', dest='delete', action='store_true',
+                        help='Delete original file after parsing')
     args = parser.parse_args()
 
-    with open(args.file, 'rb') as fd:
-        data = fd.read()
-        stream = DataStream(data, byte_order='big', is_signed=False)
-        out = OrderedDict()
-        ret, ph = generatePH(stream, out)
-        if ret == False:
-            sys.exit(1)
+    config = Config()
 
-        ret, uh = generateUH(stream, ph.creatorID, out)
-        if ret == False:
-            sys.exit(1)
+    if args.skip_plugins:
+        config.allow_plugins = False
 
-        if args.serviceable and not uh.isServiceable():
-            sys.exit(0)
+    if args.serviceable:
+        config.serviceable_only = True
 
-        if args.non_serviceable and uh.isServiceable():
-            sys.exit(0)
+    if args.non_serviceable:
+        config.non_serviceable_only = True
 
-        section_jsons = []
-        for _ in range(2, ph.sectionCount):
-            sectionID, sectionLen, versionID, subType, componentID = parserHeader(
-                stream)
-            section_json = {}
-            sectionFun(stream, section_json, sectionID, sectionLen,
-                       versionID, subType, componentID, ph.creatorID)
-            section_jsons.append(section_json)
+    if args.directory:
+        if not os.path.isdir(args.directory):
+            sys.exit(f"{args.directory} is not a valid directory")
+
+        output_dir = args.directory
+        if args.output_dir:
+            if not os.path.isdir(args.output_dir):
+                sys.exit(f"Output directory {args.output_dir} doesn't exist")
+
+            output_dir = args.output_dir
+
+        for root, _, files in os.walk(args.directory):
+            for file in files:
+                if args.extension:
+                    if args.extension != os.path.splitext(file)[1]:
```

Comment:
Nit: if args.extension and args.extension != os.path.splitext(file)[1]:

Reply:
Done.